docs - clarify sql access in sandboxing (#19450)
metabase/metabase: docs/enterprise-guide/data-sandboxes.md

![Results](images/sandboxing/advanced-example-2-results.png)

## Current limitations

### A user can only have one sandbox per table

I.e., if a user belongs to two user groups, both of which have been given sandboxed access to the same table, that user will not be able to access data from that table. You will either need to remove that user from one of those groups, or remove the sandboxed access from one of those groups.

### Data sandboxes do not support non-SQL databases

You can't currently create data sandboxes with non-SQL databases such as Google Analytics, Apache Druid, or MongoDB .

### Data sandboxes don't work for groups that can execute SQL/native queries

> If you're using data sandboxing, your groups should _not_ have access to:

Reviewer: I would say something like: "the groups with the users that you want sandboxed", to be a bit more specific

1. The native SQL editor for that data.
2. Any collections with saved SQL queries that use that data.

> The reason is that Metabase doesn't currently parse the contents of SQL queries, and therefore can't know conclusively which table(s) are referenced in a query. 

Reviewer: the contents of native SQL queries

You can't currently create data sandboxes with non-SQL databases such as Google Analytics, Apache Druid, or MongoDB .

> #### Data sandboxes do not work if users can execute SQL/native queries

Reviewer 2: ```suggestion
#### Data sandboxes don't work for groups that can execute SQL/native queries
```
Good distinction, I just substituted groups so we avoid the term "users" and emphasize that permissions are granted to groups, not people.

> If you're using data sandboxing, your user groups should _not_ have access to the native SQL editor or to saved SQL queries. The reason is that Metabase does not currently parse the contents of SQL queries, and therefore can't know conclusively which table(s) are referenced in a query. There are three specific situations to be aware of when it comes to SQL and sandboxes:

Reviewer 2: ```suggestion
If you're using data sandboxing, your groups should _not_ have access to:

1. The native SQL editor for that data.
2. Any collections with saved SQL queries that use that data.

The reason is that Metabase doesn't currently parse the contents of SQL queries, and therefore can't know conclusively which table(s) are referenced in a query. 

There are three specific situations to be aware of when it comes to SQL and sandboxes:
```

Just trying to emphasize that we give groups access to collections, not questions.

1. If a user has SQL editor access for a given database, they are able to query any table in that database, and will see unfiltered results for any table included in the query. Admins can grant SQL editor access for some databases but prohibit it on others as needed.
2. If a user views a saved SQL/native query, data sandboxes will not filter the results of that query, even if the query includes data from a table for which the user only has sandboxed access.
3. A user can start a new Custom GUI question using any saved question they have access to as the starting data for that new question (a "nested" question). If they have access to view any saved SQL questions, they can use such a question as starting data for their new question, and this new question will not be filtered by data sandboxes.

> __Note__: An important distinction to make is that you can use a saved SQL query in the _creation_ of a sandbox. [Advanced data sandboxing][advanced-sandboxing] allows you to display the results of a saved question instead of the raw table. You might choose to use a SQL query to supply the results that you'd like to display to a specific user group instead of the raw table itself. In this case, it's you as the admin who crafts the SQL query that dictates what a table looks like to a user group; people in that user group aren't viewing or executing the query itself.

Reviewer 2: ```suggestion
__Note__: An important distinction to make is that you can use a saved SQL query in the _creation_ of a sandbox. [Advanced data sandboxing][advanced-sandboxing] allows you to display the results of a saved question instead of the raw table. You might choose to use a SQL query to supply the results that you'd like to display to a specific group instead of the raw table itself. In this case, it's you as the admin who crafts the SQL query that dictates what a table looks like to a group; people in that user group aren't viewing or executing the query itself.
```
